src/Microsoft.OpenApi.CSharpAnnotations.DocumentGeneration/AssemblyLoader/CustomAssemblyLoadContext.cs

```diff
@@ -0,0 +1,166 @@
+            var assemblyName = separatorIndex > 0 ? args.Name.Substring(0, separatorIndex) : args.Name;
+
+            var assembly = TryLoadExistingAssemblyName(args.FullName);
+
+            if (assembly != null)
+            {
+                Assemblies[args.Name] = assembly;
+                return assembly;
+            }
+
+            // Try to load assembly using version.
+            var version = args.Version;
+            if (version != null)
+            {
+                var assemblyByVersion = TryLoadByVersion(AssemblyPaths, assemblyName, version.Major + "." + version.Minor + "." + version.Build + ".");
+                if (assemblyByVersion != null)
+                {
+                    Assemblies[args.Name] = assemblyByVersion;
+                    return assemblyByVersion;
+                }
+
+                assemblyByVersion = TryLoadByVersion(AssemblyPaths, assemblyName, version.Major + "." + version.Minor + ".");
+                if (assemblyByVersion != null)
+                {
+                    Assemblies[args.Name] = assemblyByVersion;
+                    return assemblyByVersion;
+                }
+
+                assemblyByVersion = TryLoadByVersion(AssemblyPaths, assemblyName, version.Major + ".");
+                if (assemblyByVersion != null)
+                {
+                    Assemblies[args.Name] = assemblyByVersion;
+                    return assemblyByVersion;
+                }
+            }
+
+            // Try to load assembly using full name, which includes version too.
+            assembly = TryLoadByAssemblyName(args.FullName);
+            if (assembly != null)
+            {
+                Assemblies[args.Name] = assembly;
+                return assembly;
+            }
+
+            // Try to load assembly using short assembly name.
+            assembly = AssemblyLoadUtility.TryLoadByName(AssemblyPaths, assemblyName);
+            if (assembly != null)
+            {
+                Assemblies[args.Name] = assembly;
+                return assembly;
+            }
+
+            Assemblies[args.Name] = TryLoadByAssemblyName(assemblyName);
+            return Assemblies[args.Name];
+        }
+
+        private Assembly TryLoadExistingAssemblyName(string assemblyName)
+        {
+            try
+            {
+                _isResolving = true;
+                return Default.LoadFromAssemblyName(new AssemblyName(assemblyName));
+            }
+            catch (Exception)
+            {
+                // Do nothing.
+            }
+            finally
+            {
+                _isResolving = false;
+            }
+
+            return null;
+        }
+
+        private Assembly TryLoadByAssemblyName(string assemblyName)
+        {
+            if (!_assembliesLoadedByAssemblyName.Contains(assemblyName))
+            {
+                try
+                {
+                    _assembliesLoadedByAssemblyName.Add(assemblyName);
```

Comment:
nit. whitespace around the ( )

Reply:
Yes i had to disable resharper due to conflicting guidelines internal and externally, will update this.

---
In reply to: [284022572](https://github.com/microsoft/OpenAPI.NET.CSharpAnnotations/pull/208#discussion_r284022572) [](ancestors = 284022572)